### PR TITLE
[FIRRTL] Alternative Connect/Invalidate Parsing

### PR DESCRIPTION
**Pull request by seldridge**
Update the FIRRTL parser to support the future, simpler-to-parse syntaxes for connect and invalidation:

    connect a, b
    invalidate c

See: https://github.com/chipsalliance/firrtl-spec/commit/40e64dcf4c82f22f18a1d3c3016528510478e1bb

### CODE REVIEW

**Review by darthscsi — COMMENTED** [review]
Look at that lack of look ahead!

LGTM.

**Review by darthscsi — APPROVED** [review]

File: lib/Dialect/FIRRTL/Import/FIRParser.cpp [hunk 1]
```diff
@@ -2622,6 +2628,33 @@ ParseResult FIRStmtParser::parseRefReleaseInitial() {
...
+ParseResult FIRStmtParser::parseConnect() {
+  auto startTok = consumeToken(FIRToken::kw_connect);
+
+  Value lhs, rhs;
+  if (parseExp(lhs, "expected connect expression") ||
+      parseExp(rhs, "expected connect expression") || parseOptionalInfo())
+    return failure();
+
+  locationProcessor.setLoc(startTok.getLoc());
+  emitConnect(builder, lhs, rhs);
```
Comment (dtzSiFive) [inline lib/Dialect/FIRRTL/Import/FIRParser.cpp:2641]: Can this check the same type-equivalence done on the other path for friendly errors?

Comment (seldridge) [reply lib/Dialect/FIRRTL/Import/FIRParser.cpp:2641]: Good idea. I'll fix this.

Comment (seldridge) [reply lib/Dialect/FIRRTL/Import/FIRParser.cpp:2641]: Fixed in: https://github.com/llvm/circt/commit/0bb6ec6212f61330aba457e29e8cc91c2f009687

**Pull request merged**